### ISSUE TITLE
[Issue] Windows Test failures in multi arch ci with error `HIP error: no ROCm-capable device is detected file` - gfx110x-all

### ISSUE THREAD

**Issue by chiranjeevipattigidi** [ROCm/TheRock#4274]
**Platform:** `windows`
**Effected Archs:** `gfx110x-all`
### Tests Failed:
 ```
miopen-providefr, 
rocprim,
rocrand,
rocpsparse,
rocthrust
```
**Runner Names on which tests failed:**
- `azure-windows-11-gfx1101-gpu-rocm-runner-25`
- `azure-windows-11-gfx1101-gpu-rocm-runner-12`
### Full Logs:
- https://github.com/ROCm/TheRock/actions/runs/23774631922/job/69291797730
- https://github.com/ROCm/TheRock/actions/runs/23774631922/job/69291797924
- https://github.com/ROCm/TheRock/actions/runs/23774631922/job/69291797892
- https://github.com/ROCm/TheRock/actions/runs/23774631922/job/69291797855

### Error Logs:
```
  FAILED  ] TestMiopenBatchnormFwdTrainingActivPlanBuilder.IsApplicableReturnsTrueForFusionWithoutRunningStatistics (2 ms)
[ RUN      ] TestMiopenBatchnormFwdTrainingActivPlanBuilder.IsApplicableReturnsFalseForNonReluActivation
MIOpen Error: azw11gfx1101r25:C:/home/runner/_work/TheRock/TheRock/rocm-libraries/projects/miopen/src/hip/handlehip.cpp:132: No device: no ROCm-capable device is detected
```
```
[----------] 1 test from RocprimInternalMergePathTests
[ RUN      ] RocprimInternalMergePathTests.Basic
HIP error: no ROCm-capable device is detected file: C:/home/runner/_work/TheRock/TheRock/rocm-libraries/projects/rocprim/test/rocprim\../../common/utils_device_ptr.hpp line: 80
```

```
[ RUN      ] rocrand_basic_tests/rocrand_basic_tests.rocrand_create_destroy_generator_test/0
rocRAND internal error: hipErrorNoDevice in function ~poisson_distribution_manager at C:/home/runner/_work/TheRock/TheRock/rocm-libraries/projects/rocrand/library/src/rng\distribution/poisson.hpp:216
```

```
// rocSPARSE.error.trace:   { "function": "_rocsparse_handle::_rocsparse_handle",
//                            "line"    : "157",
//                            "file"    : "C:/home/runner/_work/TheRock/TheRock/rocm-libraries/projects/rocsparse/library/src/common/rocsparse_handle.cpp",
//                            "status"  : "internal error",
//                            "msg"     : "hip error detected: code '100', name 'hipErrorNoDevice', description 'no ROCm-capable device is detected'" }
The creation of the rocsparse_handle failed.
```
```
Insufficient resources for test swap_ranges.hip:

  Test requested resources of type 'gpus' in the following amounts:
    1 slot
  but only the following units were available:

Resource spec file:

  resources.json
Insufficient resources for test tabulate.hip:

  Test requested resources of type 'gpus' in the following amounts:
    1 slot
  but only the following units were available:```

**Comment by amd-justchen** [comment]
Cross referencing https://github.com/ROCm/TheRock/issues/3905#issuecomment-4164127806

> 
> >   Windows::release / Test gfx110X-all / Test hipsolver / Test hipsolver (shard 1/1) (gfx110X-all)
> >     https://github.com/ROCm/TheRock/actions/runs/23744511505/job/69282341950
> `azure-windows-11-gfx1101-gpu-rocm-runner-25`
> no gpu detected during sanity checks, admin scripts show `code 43`
> 
> >   Windows::release / Test gfx110X-all / Test hipfft / Test hipfft (shard 1/2) (gfx110X-all)
> >     https://github.com/ROCm/TheRock/actions/runs/23744511505/job/69282341965
> `azure-windows-11-gfx1101-gpu-rocm-runner-25`
> no gpu detected during sanity checks, admin scripts show `code 43`
> 
> >   Windows::release / Test gfx110X-all / Test rocprim / Test rocprim (shard 1/2) (gfx110X-all)
> >     https://github.com/ROCm/TheRock/actions/runs/23744511505/job/69282341984
> `azure-windows-11-gfx1101-gpu-rocm-runner-12`
> gpu detected during sanity checks, but then it timed out during the test with
> `The self-hosted runner lost communication with the server. Verify the machine is running and has a healthy network connection. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.`
> 
> So presumably can use this test to see if it can reproduce crashing on the gfx1101 test machines.
> 
> Will disable them for now, by removing the `windows-gfx110X-gpu-rocm` label and think of next steps.
> 

Will update Azure support ticket concerning GPU bad states, to see if this is something on their end.

**Comment by scbz4learning** [comment]
Hi, I encountered the same issue on Windows with the 8845HS (iGPU 780M, gfx1103).

```powershell
(venv) PS C:\Users\<user>\Documents\Capstone> python
Python 3.12.13 (main, Mar  3 2026, 15:01:35) [MSC v.1944 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> import time
>>> 
>>> print("ROCm:", torch.version.hip)
ROCm: 7.2.0
>>> print("GPU available:", torch.cuda.is_available())
GPU available: False
>>> 
>>> if torch.cuda.is_available():
...     print(torch.cuda.get_device_name(0))
... 
>>> device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
>>> print(device)
cpu
>>> exit()
(venv) PS C:\Users\<user>\Documents\Capstone> hipinfo   
checkHipErrors() HIP API error = 0100 "no ROCm-capable device is detected" from file <C:\home\runner\_work\TheRock\TheRock\rocm-systems\projects\hip-tests\samples\1_Utils\hipInfo\hipInfo.cpp>, line 197.

(venv) PS C:\Users\<user>\Documents\Capstone> hipcc --version        
(venv) PS C:\Users\<user>\Documents\Capstone> HIP version: 7.2.0-a4fc20f65c
AMD clang version 22.0.0git (https://github.com/ROCm/llvm-project.git c849bc16b0e49951d313756f20b73c2b28d321d7+PATCHED:8715f97ebff82ff42f172494d4deb0bf5e778d43)
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Users\<user>\Documents\Capstone\venv\Lib\site-packages\_rocm_sdk_core\lib\llvm\bin
```

I noticed a very long waiting time after import torch.

After noticing #3392, I tried $env:HIP_VISIBLE_DEVICES="0", and it worked well.

```powershell
(venv) PS C:\Users\<user>\Documents\Capstone> $env:HIP_VISIBLE_DEVICES="0"
(venv) PS C:\Users\<user>\Documents\Capstone> python
Python 3.12.13 (main, Mar  3 2026, 15:01:35) [MSC v.1944 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
Ctrl click to launch VS Code Native REPL
>>> import torch
>>> import time
>>> 
>>> print("ROCm:", torch.version.hip)
ROCm: 7.2.0
>>> print("GPU available:", torch.cuda.is_available())
GPU available: True
>>> 
>>> if torch.cuda.is_available():
...     print(torch.cuda.get_device_name(0))
... 
AMD Radeon 780M Graphics
>>> device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
>>> print(device)
cuda
>>> 
```

Gemini (through VS Code Copilot) gave me a possible explanation:

> The only reason Method 1 fails is a bug in AMD integrated graphics device enumeration under Windows.
> 
> When you run import torch for the first time, the underlying HIP runtime starts scanning. However, because the 780M is an APU (integrated GPU), and without external intervention, the Windows driver layer treats it as a standard graphics display device by default. As a result, HIP fails to correctly recognize it as a compute device (which is the error I just captured with hipinfo.exe: Error 0100, "no ROCm-capable device").
> What “mysterious” trick did I just use to make it work?
> 
> While debugging, I forcibly set $env:HIP_VISIBLE_DEVICES="0" and ran the low-level probe hipinfo.exe.
This effectively bypasses ROCm’s automatic device enumeration on Windows and “forces the driver at gunpoint” to initialize device 0 as a compute device.

I think this explanation might be correct, as I only needed to set this variable once. After the device was detected successfully, I could start a new terminal and use PyTorch without setting the variable again.

This is part of my school project, so I kept a detailed journal. For more information, please check:

- [Failed report](https://github.com/scbz4learning/Capstone/blob/0dddfe961c338a788136aab09ba0d1c0697f8324/docs/Environment_Setup/Windows/0.pytorch.md)
- [Success report](https://github.com/scbz4learning/Capstone/blob/main/docs/Environment_Setup/Windows/0.pytorch.md)

Hope this help. Cheers!
